packages/amplify_datastore_plugin_interface/lib/src/types/temporal/temporal_date.dart

```diff
@@ -0,0 +1,131 @@
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:date_time_format/date_time_format.dart';
+
+import 'temporal.dart';
+
+/// Represents a valid extended ISO-8601 Date string, with an optional timezone offset.
+/// YYYY-MM-DD±hh:mm:ss  (ISO_OFFSET_DATE)
+/// or
+/// YYYY-MM-DD (ISO_LOCAL_DATE)
+/// https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html#appsync-defined-scalars
+class TemporalDate {
+  DateTime _dateTime;
+  Duration _offset;
+
+  /// Constructs a new TemporalDate at the current date.
+  static TemporalDate now() {
+    return TemporalDate(DateTime.now());
+  }
+
+  /// Constructs a new TemporalDate from a Dart DateTime
+  /// The time fields (hour, minute, second, etc) are ignored
+  TemporalDate(DateTime dateTime) {
+    dateTime = dateTime.toUtc();
+    _dateTime = DateTime.utc(
+      dateTime.year,
+      dateTime.month,
+      dateTime.day,
+    );
+  }
+
+  /// Constructs a new TemporalDate from a Dart DateTime and Duration
+  /// The time fields (hour, minute, second, etc) are ignored
+  TemporalDate.withOffset(DateTime dateTime, Duration offset) {
+    dateTime = dateTime.toUtc();
+    _dateTime = DateTime.utc(
+      dateTime.year,
+      dateTime.month,
+      dateTime.day,
+    );
+
+    if (offset.inDays > 0) {
+      throw new Exception("Cannot have an offset in days (hh:mm:ss)");
+    }
+
+    _offset = offset;
+  }
+
+  /// Constructs a new TemporalDate from a ISO8601 string adhering to the format:
+  ///   YYYY-MM-DD
+  ///   YYYY-MM-DDZ
+  ///   without Z:
+  ///     +hh:mm
+  ///     +hh:mm:ss
+  TemporalDate.fromString(String iso8601String) {
+    // TODO: enforce month 1-12
+    RegExp regExp = new RegExp(
+        r'^([0-9]{4}-[0-1][0-9]-[0-3][0-9])((z|Z)|((\+|-)[0-2][0-9]:[0-5][0-9](:[0-5][0-9])?))?',
+        caseSensitive: false,
+        multiLine: false);
+
+    // Validate
+    String regexString = regExp.stringMatch(iso8601String);
+    if (regexString != iso8601String) {
+      throw new Exception("invalid string input");
+    }
+
+    // Remove Z
+    regexString = iso8601String.replaceAll(new RegExp(r'(z|Z)'), '');
+
+    // Extract Date
+    var match = regExp.matchAsPrefix(regexString);
+
+    // Parse cannot take a YYYY-MM-DD as UTC!
+    DateTime dateTime = DateTime.parse(match.group(1));
+    _dateTime = DateTime.utc(dateTime.year, dateTime.month, dateTime.day);
+
+    // Extract Offset
+    if (match.group(2) != null && match.group(2).isNotEmpty)
+      _offset = Temporal.stringToOffset(match.group(2));
+    else if (iso8601String.toLowerCase().contains("z")) _offset = Duration();
+  }
+
+  /// Return offset
+  Duration getOffset() {
+    return _offset;
+  }
+
+  /// Return DateTime
+  DateTime getDateTime() {
+    return _dateTime;
+  }
```

Comment:
We should rename this as `getDateTimeInUtc()` since we are manipulating what customers provided and converting the provided datetime to UTC, it's not really the same what they provided.

```dart
var myDate = DateTime.now()
var temporalDate = TemporalDate(myDate);

print(myDate == temporalDate.getDateTime()); // returns false;
```

Ref: https://api.dart.dev/stable/2.10.5/dart-core/DateTime/operator_equals.html